src/main/java/com/bart/scorebetlive442/repository/TeamRepository.java

```diff
@@ -2,9 +2,18 @@
 
 import com.bart.scorebetlive442.entity.TeamEntity;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
 import org.springframework.stereotype.Repository;
 
+import java.util.List;
+import java.util.Set;
+
 @Repository
 public interface TeamRepository extends JpaRepository<TeamEntity, Long> {
-    TeamEntity findByCountryIgnoreCase(String country);
+    @Query("SELECT t FROM TeamEntity t WHERE t.id IN :teamIds AND (t.leagueEntity IS NULL OR t.leagueEntity.id != :leagueId)")
+    List<TeamEntity> findTeamsByIdsAndNotInLeague(@Param("teamIds") Set<Long> teamIds, @Param("leagueId") Long leagueId);
+
+    @Query("SELECT t FROM TeamEntity t WHERE t.id IN :teamIds AND t.leagueEntity.id = :leagueId")
+    List<TeamEntity> findTeamsByIdsAndInLeague(@Param("teamIds") Set<Long> teamIds, @Param("leagueId") Long leagueId);
```

Comment:
(info) teoretycznie jeżeli nazwa paremetru jest taka sama jak w Query to nie trzeba dawać @Param 
(info) poczytaj o JPA Projections dodatkowo https://www.baeldung.com/spring-data-jpa-projections
https://docs.spring.io/spring-data/jpa/reference/repositories/projections.html
https://vladmihalcea.com/spring-jpa-dto-projection/